core/chains/evm/chain.go

```diff
@@ -142,22 +142,39 @@ type AppConfig interface {
 
 type ChainRelayExtenderConfig struct {
 	Logger   logger.Logger
-	DB       *sqlx.DB
 	KeyStore keystore.Eth
-	*RelayerConfig
+	*ChainOpts
 }
 
-// options for the relayer factory.
-// TODO BCF-2508 clean up configuration of chain and relayer after BCF-2440
-// the factory wants to own the logger and db
-// the factory creates extenders, which need the same and more opts
-type RelayerConfig struct {
+func (c ChainRelayExtenderConfig) Validate() error {
+	var err error
+	if c.Logger == nil {
+		err = errors.Join(err, fmt.Errorf("nil Logger"))
```

Comment:
I'm surprised that vet and the linters don't care about this:
```suggestion
		err = errors.Join(err, errors.New("nil Logger"))
```